loki/subroutine.py

```diff
@@ -467,6 +467,12 @@ def enrich_calls(self, routines):
                         name_type.dtype.procedure is not routine  # ProcedureType not linked to routine
                     )
                     if update_symbol:
+                        # Remove existing symbol from symbol table if defined in interface block
+                        for node in [node for intf in self.interfaces for node in intf.body]:
+                            if getattr(node, 'name', None) == call.name:
+                                if node.parent == self:
+                                    node.parent = None
```

Comment:
This breaks with the nested scoping behaviour that we're using normally. However, the root problem is that interface blocks shouldn't really have a parent scope in the first place because they can't inherit symbols,  so I'm ok with this temporary solution.

Reply:
Filed a follow-up ticket in #147